### ISSUE TITLE
.zinit-load-snippet/.zinit-load errors after doing a self-update

### ISSUE THREAD

**Issue by rifen** [zdharma/zinit#386]
**Describe the bug**
The original issue was zsh was hitting 100% cpu usage if I left my session open. Then I started messing with things and in the end I did a zinit self-update and my .zshrc broke at that point and I haven't really been able to understand why. This is what I get on start-up:
```
> .zinit-load:11: closing brace expected
> .zinit-load:15: invalid function definition
> .zinit-load:11: closing brace expected
> .zinit-load:15: invalid function definition
> .zinit-load:11: closing brace expected
> .zinit-load:15: invalid function definition
> .zinit-load:11: closing brace expected
> .zinit-load:15: invalid function definition
> .zinit-load:11: closing brace expected
> .zinit-load:15: invalid function definition
> .zinit-load:11: closing brace expected
> .zinit-load:15: invalid function definition
> ~ >                                                                    13:40:29
> .zinit-load-snippet:11: closing brace expected
> .zinit-load-snippet:37: invalid function definition
> ~ >                                                                    13:40:29
> .zinit-load-snippet:11: closing brace expected
> .zinit-load-snippet:37: invalid function definition
> ~ >                                                                    13:40:29
> .zinit-load-snippet:11: closing brace expected
> .zinit-load-snippet:37: invalid function definition
> ~ >                                                                    13:40:29
> .zinit-load-snippet:11: closing brace expected
> .zinit-load-snippet:37: invalid function definition
> ~ >                                                                    13:40:29
> .zinit-load-snippet:11: closing brace expected
> .zinit-load-snippet:37: invalid function definition
> ~ >                                                                    13:40:29
> .zinit-load-snippet:11: closing brace expected
> .zinit-load-snippet:37: invalid function definition
> ~ >                                                                    13:40:29
> .zinit-load-snippet:11: closing brace expected
> .zinit-load-snippet:37: invalid function definition
> ~ >                                                                    13:40:29
> .zinit-load-snippet:11: closing brace expected
> .zinit-load-snippet:37: invalid function definition
> ~ >                                                                    13:40:29
> .zinit-load:11: closing brace expected
> .zinit-load:15: invalid function definition
> ~ >                                                                    13:40:29
> .zinit-load:11: closing brace expected
> .zinit-load:15: invalid function definition
> ~ >                                                                    13:40:29
```
Looked for syntax changes within zinit itself and saw none. Re-installed everything and still received these issues

**To Reproduce**
Ran my uninstall/install scripts.

**Expected behavior**
Before the self-update everything worked soundly.

**Screenshots**
Zinit portion of my .zshrc
```
##"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
#""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
##"""                                Zinit                                   """
##"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
### Added by Zinit's installer
if [[ ! -f $HOME/.zinit/bin/zinit.zsh ]]; then
    print -P "%F{33}▓▒░ %F{220}Installing DHARMA Initiative Plugin Manager (zdharma/zinit)…%f"
    command mkdir -p $HOME/.zinit
    command git clone https://github.com/zdharma/zinit $HOME/.zinit/bin && \
        print -P "%F{33}▓▒░ %F{34}Installation successful.%f" || \
        print -P "%F{160}▓▒░ The clone has failed.%f"
fi
source "$HOME/.zinit/bin/zinit.zsh"
autoload -Uz _zinit
(( ${+_comps} )) && _comps[zinit]=_zinit
### End of Zinit installer's chunk
#
# zinit plugins
#

# OMZ
zinit wait lucid for \
      OMZ::lib/compfix.zsh \
      OMZ::lib/functions.zsh \
      OMZ::lib/git.zsh \
      OMZ::lib/termsupport.zsh \
      OMZ::plugins/aws/aws.plugin.zsh \
      OMZ::plugins/command-not-found/command-not-found.plugin.zsh \
      OMZ::plugins/git/git.plugin.zsh \
      OMZ::plugins/sudo/sudo.plugin.zsh

#LS_COLORS
zinit ice atclone"dircolors -b LS_COLORS > c.zsh" atpull'%atclone' pick"c.zsh" nocompile'!'
zinit light trapd00r/LS_COLORS

#zsh-256color
zinit light chrissicool/zsh-256color

#agkozak-zsh-prompt
zinit light agkozak/agkozak-zsh-prompt

#zsh-fast-alias-tips
zinit ice from'gh-r' as'program'
zinit light sei40kr/fast-alias-tips-bin
zinit light sei40kr/zsh-fast-alias-tips

# Autosuggestions & fast-syntax-highlighting
zinit ice wait lucid atinit"ZINIT[COMPINIT_OPTS]=-C; zpcompinit; zpcdreplay"
zinit light zdharma/fast-syntax-highlighting

# zsh-autosuggestions
zinit ice wait lucid atload"!_zsh_autosuggest_start"
zinit load zsh-users/zsh-autosuggestions

#tldr man pages
zinit ice wait"2" as"program" pick"tldr" lucid
zinit light raylee/tldr

#diff-so-fancy
zinit ice wait"2" lucid as"program" pick"bin/git-dsf"
zinit light zdharma/zsh-diff-so-fancy

#fzf
zplugin pack for fzy

# OMZ Customizations

#agkozak-zsh-prompt CUSTOMIZATION
AGKOZAK_COLORS_EXIT_STATUS=blue
AGKOZAK_COLORS_USER_HOST=grey
AGKOZAK_COLORS_PATH=blue
AGKOZAK_COLORS_BRANCH_STATUS=250
AGKOZAK_COLORS_PROMPT_CHAR=default      # Default text color
AGKOZAK_COLORS_CMD_EXEC_TIME=default    # Default text color
AGKOZAK_COLORS_VIRTUALENV=green
AGKOZAK_MULTILINE=0
AGKOZAK_PROMPT_CHAR=( '%F{magenta}>%f' '%F{red}>%f' '%F{magenta}<%f' )
AGKOZAK_LEFT_PROMPT_ONLY=1
AGKOZAK_CUSTOM_RPROMPT+='%F{32}%*'
AGKOZAK_PROMPT_DIRTRIM=2
AGKOZAK_USER_HOST_DISPLAY=0

FAST_ALIAS_TIPS_PREFIX="Seth use: \033[0;31m"
```

**Versions:**
 - $ZSH_VERSION == 5.0.2 (RHEL doesn't offer anything higher)
 - Red Hat Enterprise Linux Server release 7.8 (Maipo)

**Additional context**
None that I can think of, but feel free to ask.

**Comment by psprint** [comment]
Can you do: `git -C ~/.zinit/bin pull` and see if it helps?

**Comment by rifen** [comment]
1. Looks like it did something but nothing has changed.
```
 README.md          |  2 +-
 zinit-autoload.zsh | 18 +++++++++---------
 zinit-install.zsh  | 34 +++++++++++++++++-----------------
 zinit.zsh          | 10 +++++-----
```

2. I updated just yesterday, in the afternoon time.
3. 81f4836e5015bfbc3eba2f262b1f84910f3ba758

**Comment by horta** [comment]
I have a related problem here on Red Hat 7.4, zsh 5.0.2 (x86_64-redhat-linux-gnu)

First uninstalled it: rm -rf ~/.zinit then installed again

```
[horta@noah-login-02]~% sh -c "$(curl -fsSL https://raw.githubusercontent.com/zdharma/zinit/master/doc/install.sh)"

▓▒░ Installing DHARMA Initiative Plugin Manager at /homes/horta/.zinit/bin

Cloning into 'bin'...
◢ ███████████ OBJ: 100, PACK: 19632/19632, COMPR: 100%, REC: 100%, RES: 100%  

▓▒░ Zinit succesfully installed at /homes/horta/.zinit/bin.
▓▒░ Version: v3.1-372-g1d85dbce
▓▒░ Updating /homes/horta/.zshrc (10 lines of code, at the bottom)

▓▒░ Would you like to add 4 useful plugins - the most useful annexes (Zinit extensions that add new functions-features to the plugin manager) to the zshrc as well? It will be the following snippet:

# Load a few important annexes, without Turbo
# (this is currently required for annexes)
zinit light-mode for \
    zinit-zsh/z-a-rust \
    zinit-zsh/z-a-as-monitor \
    zinit-zsh/z-a-patch-dl \
    zinit-zsh/z-a-bin-gem-node

▓▒░ Enter y/n and press Return: y

▓▒░ Done.


▓▒░ A quick intro to Zinit: below are all the available Zinit
▓▒░ ice-modifiers, grouped by their role by different colors): 
▓▒░
▓▒░ id-as'' as'' from'' wait'' trigger-load'' load'' unload'' 
▓▒░ pick'' src'' multisrc'' pack'' param'' extract'' atclone''
▓▒░ atpull'' atload'' atinit'' make'' mv'' cp'' reset''
▓▒░ countdown'' compile'' nocompile'' nocd'' if'' has'' 
▓▒░ cloneopts'' depth'' proto'' on-update-of'' subscribe''
▓▒░ bpick'' cloneonly'' service'' notify'' wrap-track''
▓▒░ bindmap'' atdelete'' ver'' 
 
▓▒░ No-value (flag-only) ices:
▓▒░ svn git silent lucid light-mode is-snippet blockf nocompletions
▓▒░ run-atpull reset-prompt trackbinds aliases sh bash ksh csh

For more information see:
- README section on the ice-modifiers:
    - https://github.com/zdharma/zinit#ice-modifiers,
- intro to Zinit at the Wiki:
    - https://zdharma.org/zinit/wiki/INTRODUCTION/,
- zinit-zsh GitHub account, which holds all the available Zinit annexes:
    - https://github.com/zinit-zsh/,
- For-Syntax article on the Wiki; it is less directly related to the ices, however, it explains how to use them conveniently:
    - https://zdharma.org/zinit/wiki/For-Syntax/.
[horta@noah-login-02]~% 
```


Exited the ssh session and logged back in. That is what I got:

```
% ssh ebi-login                                                                                                                                                                                   
Last login: Thu Jul 30 03:36:37 2020 from ligate.ebi.ac.uk
.zinit-load-object:2: bad pattern: __opt=( zinit-zsh/z-a-rust
[horta@noah-login-02]~% 
```

**Comment by horta** [comment]
I guess the problem is here: https://github.com/zdharma/zinit/blob/1d85dbceb0ca3e19c3bf32899b2c7a388c3c6dff/zinit.zsh#L1203

That functio .zinit-load-object is not working here, not even by just doing:

```
zinit light miekg/lean
```

It complains

```
.zinit-load-object:2: unknown file attribute
```

**Comment by horta** [comment]
Maybe relevant: https://stackoverflow.com/questions/37220495/zsh-unknown-file-attribute

**Comment by horta** [comment]
I had to rollback to commit ec8a0a5a9f43b91d1ffbd5bbef199525392cf063
It works perfectly. I guess you added some features not supported by zsh 5.0.x

Let me know how can I help with it.

**Comment by psprint** [comment]
The bug in `*load-object` has been fixed.

**Comment by rifen** [comment]
Hm maybe I didn't do the correct procedure. I started by running:
`zi self-update`
Then I ran the command you gave before:
`git -C ~/.zinit/bin pull`
Got a already up to date.

`git -C ~/.zinit/bin rev-parse HEAD`:
534083182e21df0b9c8dd34246cc39217cc727f6

**Comment by horta** [comment]
Thanks, @psprint 
It looks like there is still a zsh-dependent syntax with the latest version:

```
.zinit-load:11: closing brace expected
.zinit-load:15: invalid function definition
.zinit-load:11: closing brace expected
.zinit-load:15: invalid function definition
.zinit-load:11: closing brace expected
.zinit-load:15: invalid function definition
.zinit-load:11: closing brace expected
.zinit-load:15: invalid function definition
.zinit-load:11: closing brace expected
.zinit-load:15: invalid function definition
.zinit-load:11: closing brace expected
.zinit-load:15: invalid function definition
.zinit-load:11: closing brace expected
.zinit-load:15: invalid function definition
```

**Comment by horta** [comment]
It seems to be this line: https://github.com/zdharma/zinit/blob/534083182e21df0b9c8dd34246cc39217cc727f6/zinit.zsh#L1483

**Comment by horta** [comment]
If I do
```
echo "${functions[+zinit-message]}"
```

right before that line I get:


```
.zinit-load:15: bad pattern: \tbuiltin emulate -LR zsh -o extendedglob\n\tif [[ $1 = -* ]]\n\tthen\n\t\tlocal opt=$1\n\t\tshift\n\telse\n\t\tlocal opt\n\tfi\n\tlocal -a msg\n\tmsg=(${${@//(#b)([\\]([\{]([^\}]##)[\}])|([\{]([^\}]##)[\}])([^\{\\]#))/$match[2]${${functions[.zinit-formatter-$match[5]]:+${$(.zinit-formatter-$match[5] "$match[6]"; builtin print -rn -- $REPLY):-\M-b^]\M-b^>\M-b^<\M-b^`}}:-${ZINIT[col-$match[5]]-$match[4]}$match[6]}}//\M-b^]\M-b^>\M-b^<\M-b^`}) \n\tbuiltin print -Pr ${opt:#--} -- ${msg[1]:#--} ${msg[2,-1]}
```

**Comment by horta** [comment]
Hi @psprint , since I cannot make sense of the error, I've got a set of commands that you can use to replicate the problem using docker:

```
docker run -it centos:7.4.1708 /bin/bash
yum install zsh git -y
sh -c "$(curl -fsSL https://raw.githubusercontent.com/zdharma/zinit/master/doc/install.sh)"
zsh
```

**Comment by rifen** [comment]
Hope this can be fixed still. I ended up compiling zsh from source on my RHEL station and it works just fine now. I was having other issues with zsh but it's fixed now.

**Comment by rhuang2014** [comment]
I got a slightly different error after I ran `zi self-update && zi update -r` today.
```
.zinit-load-object:12: number truncated after 20 digits: 010000000000000000000
.zinit-load-object:12: number truncated after 21 digits: 0100000000000000000000
.zinit-load-object:12: number truncated after 21 digits: 01000000000000000000000
.zinit-load-object:12: number truncated after 21 digits: 010000000000000000000000
.zinit-load-object:12: number truncated after 21 digits: 0100000000000000000000000
.zinit-load-object:12: number truncated after 21 digits: 01000000000000000000000000
.zinit-load-object:12: number truncated after 21 digits: 010000000000000000000000000
.zinit-load-object:12: number truncated after 21 digits: 0100000000000000000000000000
.zinit-load-object:12: number truncated after 21 digits: 01000000000000000000000000000
.zinit-load-object:12: number truncated after 21 digits: 010000000000000000000000000000
.zinit-load-object:12: number truncated after 21 digits: 0100000000000000000000000000000
```

**Comment by horta** [comment]
This merge solves the problem for me: https://github.com/zdharma/zinit/pull/414

**Comment by bagelbits** [comment]
Looks like this is still failing:
```
.zinit-load:11: closing brace expected
.zinit-load:16: invalid function definition
.zinit-load:11: closing brace expected
.zinit-load:16: invalid function definition
.zinit-load:11: closing brace expected
.zinit-load:16: invalid function definition
.zinit-load:11: closing brace expected
.zinit-load:16: invalid function definition
.zinit-load:11: closing brace expected
.zinit-load:16: invalid function definition
```

**Comment by bagelbits** [comment]
Looks like this is failing again:
```
.zinit-load:11: closing brace expected
.zinit-load:16: invalid function definition
.zinit-load:11: closing brace expected
.zinit-load:16: invalid function definition
.zinit-load:11: closing brace expected
.zinit-load:16: invalid function definition
.zinit-load:11: closing brace expected
.zinit-load:16: invalid function definition
.zinit-load:11: closing brace expected
.zinit-load:16: invalid function definition
```

**Comment by psprint** [comment]
Could you update (`git -C ~/.zinit/bin pull`) and check again, and if the issue persists, then please send the result of:

```zsh
autoload allopt
allopt options.txt
```
?

**Comment by bagelbits** [comment]
This is specifically an issue with zsh 5.0.2. Which I completely understand if you're not still supporting.

**Comment by bagelbits** [comment]
It also looks like someone re-added the character that @horta removed in their PR.

**Comment by xbb** [comment]
I found out that if I replace that characters with the unicode escape sequence `"\u2190\u2191\u2192"` zinit loads without errors.

However if I print the function with `echo "${functions[+zinit-message]}"` zsh prints the error `zsh: character not in range`

Executing the same echo command with `LC_CTYPE=en_US.UTF-8` instead of the default `C.UTF-8` works fine.

Also when `LC_CTYPE` is `C.UTF-8` I cannot paste that characters at all in the terminal

To recap with `C.UTF-8`:

* Can't paste `←↑→` in the terminal

* `echo "\u2190\u2191\u2192"` fails with character out of range

* `echo $'\xe2\x86\x90\xe2\x86\x91\xe2\x86\x92'` works fine

With `LC_CTYPE=en_US.UTF-8`:

All above works and can paste `←↑→` in the terminal

I'm using RedHat Atomic Host 7.9 and `bash` doesn't seem to be affected

**Comment by psprint** [comment]
I've pushed a patch that removes the problematic char `↑`. Could you update and check?

**Comment by xbb** [comment]
It loads without errors, but `echo "${functions[+zinit-message]}"` shows "bad pattern"

```
zsh: bad pattern: \tbuiltin emulate -LR zsh -o extendedglob\n\tif [[ $1 = -* ]]\n\tthen\n\t\tlocal opt=$1\n\t\tshift\n\telse\n\t\tlocal opt\n\tfi\n\tlocal -a msg\n\tmsg=(${${@//(#b)([\\]([\{]([^\}]##)[\}])|([\{]([^\}]##)[\}])([^\{\\]#))/$match[2]${${functions[.zinit-formatter-$match[5]]:+${$(.zinit-formatter-$match[5] "$match[6]"; builtin print -rn -- $REPLY):-\M-b^]\M-b^<}}:-$(.zinit-main-message-formatter "$match[4]" "$match[5]" "$match[6]"; \\n  builtin print -rn -- "$REPLY")}}//\M-b^]\M-b^<}) \n\tbuiltin print -Pr ${opt:#--} -- ${msg[1]:#--} ${msg[2,-1]}
```

Also `zinit update` is broken:

```
❯ zinit update
Assuming --all is passed
Updating Zinit…
Note: updating also unloaded plugins
Updating agkozak/zsh-z
Unknown option: -C
usage: git [--version] [--help] [-c name=value]
           [--exec-path[=<path>]] [--html-path] [--man-path] [--info-path]
           [-p|--paginate|--no-pager] [--no-replace-objects] [--bare]
           [--git-dir=<path>] [--work-tree=<path>] [--namespace=<name>]
           <command> [<args>]
```

.zshrc:

```
zinit wait lucid for \
  atinit"ZINIT[COMPINIT_OPTS]=-C; zicompinit; zicdreplay" \
    zdharma/fast-syntax-highlighting
```

**Comment by psprint** [comment]
Update should now work, thanks.
